pkg/cluster/fixssh.go

```diff
@@ -82,22 +112,34 @@ func (m *manager) fixSSH(ctx context.Context) error {
 			}
 		}
 	}
+	return nil
+}
 
+func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup, nicName string, nic *mgmtnetwork.Interface) error {
+	ipc := (*nic.InterfacePropertiesFormat.IPConfigurations)[0]
```

Comment:
Please can we defend against `*nic.InterfacePropertiesFormat.IPConfigurations` being invalid here with a check for:
```go
if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
	return fmt.Errorf("unable to remove backendpools from NIC as there are no IP configurations for %s in resource group %s", nicName, resourceGroup)
}
```

Reply:
Updated check for NIC IP configurations validation.